tests/functional/ecd/test_ecd.py

```diff
@@ -0,0 +1,135 @@
+        self.ecd_url = 'http://' + self.ecd_addr
+        if len(self.conf['services']['rawx']) < 9:
+            self.skipTest("need at least 9 rawx to run")
+
+    def _download_metachunk(self, meta, chunks):
+        headers = dict()
+        headers[CHUNK_METADATA_PREFIX + "content-chunk-method"] = \
+            meta['chunk_method']
+        headers[CHUNK_METADATA_PREFIX + "chunk-size"] = \
+            meta['chunk_size']
+        for chunk in chunks:
+            headers[CHUNK_METADATA_PREFIX + "chunk-" + str(chunk['num'])] = \
+                chunk['real_url']
+        resp = self.request('GET', self.ecd_url, headers=headers)
+        return resp
+
+    def _test_download(self, length):
+        container = random_str(8)
+        obj = random_str(8)
+        expected_data = random_data(length)
+        chunks, _, _, meta = self.storage.object_create_ext(
+            self.account, container, obj_name=obj, data=expected_data,
+            policy='EC')
+        storage_method = STORAGE_METHODS.load(meta['chunk_method'])
+        sorted_chunks = _sort_chunks(chunks, storage_method.ec)
+
+        data = ''
+        for pos in range(len(sorted_chunks)):
+            resp = self._download_metachunk(meta, sorted_chunks[pos])
+            self.assertEqual(200, resp.status)
+            data += resp.data
+        self.assertEqual(expected_data, data)
+
+    def test_download_0(self):
+        self._test_download(0)
+
+    def test_download_1K(self):
+        self._test_download(1024)
+
+    def test_download_10K(self):
+        self._test_download(1024*10)
+
+    def test_download_100K(self):
+        self._test_download(1024*100)
+
+    def test_download_1M(self):
+        self._test_download(1024*1024)
+
+    def test_download_10M(self):
+        self._test_download(1024*1024*10)
+
+    def test_download_with_missing_chunks(self):
+        container = random_str(8)
+        obj = random_str(8)
+        expected_data = random_data(10)
+        chunks, _, _, meta = self.storage.object_create_ext(
+            self.account, container, obj_name=obj, data=expected_data,
+            policy='EC')
+        storage_method = STORAGE_METHODS.load(meta['chunk_method'])
+        sorted_chunks = _sort_chunks(chunks, storage_method.ec)
+
+        for i in range(storage_method.ec_nb_parity):
+            data = ''
+            for pos in range(len(sorted_chunks)):
+                chunk = random.choice(sorted_chunks[pos])
+                sorted_chunks[pos].remove(chunk)
+                resp = self._download_metachunk(meta, sorted_chunks[pos])
+                self.assertEqual(200, resp.status)
+                data += resp.data
+            self.assertEqual(expected_data, data)
+
+        for pos in range(len(sorted_chunks)):
+            chunk = random.choice(sorted_chunks[pos])
+            sorted_chunks[pos].remove(chunk)
+            resp = self._download_metachunk(meta, sorted_chunks[pos])
+            self.assertEqual(500, resp.status)
```

Comment:
It is not related to this PR but IMHO we should use another error code instead of this generic 500

Reply:
`oio-swift` returns the same error code (500) in this same case.
What error code do you suggest?